racket/src/racket/src/jitprep.c

```diff
@@ -391,7 +391,6 @@ static Scheme_Object *with_immed_mark_jit(Scheme_Object *o)
 
 Scheme_Object *scheme_case_lambda_jit(Scheme_Object *expr)
 {
-#ifdef MZ_USE_JIT
   Scheme_Case_Lambda *seqin = (Scheme_Case_Lambda *)expr;
 
   if (!seqin->native_code) {
```

Comment:
The `native_code` field exists only when `MZ_USE_JIT` is defined, so that's why the `#ifdef` is needed around here.